x-pack/plugins/ml/server/routes/schemas/alerting_schema.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+
+export const mlAnomalyDetectionAlertParams = schema.object({
+  jobSelection: schema.object(
+    {
+      jobIds: schema.maybe(schema.arrayOf(schema.string())),
```

Comment:
We avoid using`schema.maybe()` because it can cause partial update issues.  If the property can be undefined, and has a previous value stored, but is then updated with that property not set (ie, set to `undefined`), then the previous property value won't be updated.  We use `schema.nullable()` instead, which will only allow `null` values (during the schema validation).

Alternatively, I think you could use `schema.arrayOf(schema.string(), { default: [] })`, if that's legal, which would always provide an array value, whether it's undefined or not.

Reply:
Changed the schema in ce50bd2eb06e6958d58cc1ae598a47785f2a64aa.
cc @peteharverson and everyone who tested this, you'll need to update your existing alerts (simply edit and save) so the `shema` applies the default value.